kubernetes/base/dynamic/client.py

```diff
@@ -195,10 +195,13 @@ def watch(self, resource, namespace=None, name=None, label_selector=None, field_
         """
         if not watcher: watcher = watch.Watch()
 
+        # Use field selector to query for named instance so the watch parameter is handled properly.
+        if name:
+            field_selector = f"metadata.name={name}"
```

Comment:
If in addition to `name`, the user also specified `field_selector` when calling `watch()`, shall we also persist the user-specified value instead of overwriting it?

Reply:
I thought about that, but according to this: https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/#chained-selectors multiple field selectors are and-ed together, and since `name` is unique including additional field selectors didn't seem to make any sense.

Reply:
Sorry for the super late response. That makes sense. Thanks!